addons/crm/static/src/views/crm_kanban/crm_kanban_model.js

```diff
@@ -0,0 +1,59 @@
+/** @odoo-module **/
+
+import { KanbanModel } from "@web/views/kanban/kanban_model";
+import { checkRainbowmanMessage } from "@crm/views/check_rainbowman_message";
+
+export class CrmKanbanModel extends KanbanModel {
+    setup(params, { orm, effect }) {
+        super.setup(...arguments);
+        this.ormService = orm;
+        this.effect = effect;
+    }
+}
+
+export class CrmKanbanDynamicGroupList extends CrmKanbanModel.DynamicGroupList {
+    getFieldsToAggregate() {
+        const result = super.getFieldsToAggregate();
+        if (this.model.progressAttributes.recurring_revenue_sum_field) {
+            result.push(this.model.progressAttributes.recurring_revenue_sum_field.name);
+        }
+        return result;
+    }
+
+    /**
+     * @override
+     *
+     * If the kanban view is grouped by stage_id check if the lead is won and display
+     * a rainbowman message if that's the case.
+     */
+    async moveRecord(dataRecordId, dataGroupId, refId, targetGroupId) {
+        const res = await super.moveRecord(...arguments);
+        const sourceGroup = this.groups.find((g) => g.id === dataGroupId);
+        const targetGroup = this.groups.find((g) => g.id === targetGroupId);
+        if (
+            dataGroupId !== targetGroupId &&
+            sourceGroup &&
+            targetGroup &&
+            sourceGroup.groupByField.name === "stage_id"
+        ) {
+            const record = targetGroup.list.records.find((r) => r.id === dataRecordId);
+            await checkRainbowmanMessage(this.model.ormService, this.model.effect, record.resId);
+        }
+        return res;
+    }
+}
+
+export class CrmKanbanGroup extends CrmKanbanModel.Group {
+    updateAggregates(groupData) {
+        super.updateAggregates(...arguments);
+        const rrField = this.model.progressAttributes.recurring_revenue_sum_field;
+        const group = groupData.find(g => this.valueEquals(g[this.groupByField.name]));
```

Comment:
`groupByField` is the current group by indeed.
It's a singular group as is `this`, basically `groupData` is the result of the `web_search_group`, this method is called whenever groups have to be reloaded to update some data, for instance if you have a progressbar it will update the `sum_field`; here we do the same but for the recurring revenue field :)